src/Microsoft.SymbolStore.Client/SymbolCache.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.SymbolStore.Client
+{
+    public class SymbolCache
+    {
+        private string _location;
+
+        public SymbolCache(string location)
+        {
+            if (location == null)
+                throw new ArgumentNullException(nameof(location));
+
+            if (string.IsNullOrWhiteSpace(location))
+                throw new ArgumentException($"Argument '{nameof(location)}' cannot be an empty string.");
+
+            _location = location;
+            if (!Directory.Exists(_location))
+                Directory.CreateDirectory(location);
+        }
+
+        #region PDB Functions
+        public string GetPdbFromCache(string pdbName, Guid guid, int age)
```

Comment:
If the cache isn't frequently used directly by consumers of the library, it may not be worthwhile creating a strongly typed set of Get/Store/Remove/GetCacheLocation methods for every file type. Instead you can could create a generic set of methods that takes a relative path and let the caller calculate it for you.